foundry/applications/dialog.d.ts

```diff
@@ -235,10 +235,10 @@ declare class Dialog extends Application {
   }?: {
     title: string;
     content: string;
-    label: string;
-    callback: (html: JQuery) => T;
-    render: (html: JQuery) => void;
-    options: Partial<Dialog.Options> & { jQuery: true };
+    label?: string;
+    callback?: (html: JQuery) => T;
```

Comment:
`callback` must be present because it is called when the button is clicked. there is no check if it is falsy. So if you provide undefined here, you will get an Uncaught ReferenceError when the button is clicked.

I think the other parameters are technically optional, though the foundry does not consider them optional. @kmoschcau what do you think?

All of this also applies to the other overloads.

Reply:
Yes callback should not be optional, I mistakenly set it as such. The `render` function is indeed optional, as Foundry explicitly checks if it is a defined Function before trying to call it, and Foundry itself does not typically include the `render` option when invoking the Dialog. `label` is also optional, as it is used as the button text, which may be blank. I think the expectation is that you can have it blank if you provide an icon, but neither is strictly required by the API implementation. Also, the `options` field is optional as it is the `Application.Options` value that is passed on to the base constructor which is always an optional argument (Application has a default value for the options argument).